src/executor/executor.c

```diff
@@ -5257,7 +5335,18 @@ int main(int argc, char *argv[]) {
                 goto out;
         }
 
-        arg_serialization = safe_fclose(arg_serialization);
+        /* The worker might have been spawned long before it needs to execute a child, so clear the
+         * environment and set it to what the manager says it should be */
+        clearenv();
+        STRV_FOREACH(e, context.manager_environment) {
+                r = putenv(*e);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
can you make this a helper call? set_full_environment() or so? and put in env-util.[ch]?